contracts/USSLib.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+// Contains structs and functions used by SpokePool contracts to facilitate universal settlement.
+library USSLib {
```

Comment:
So the library is just a way for us to namespace all of this so it's clear what the "old version" was and what the new version is?

Alternatively, we could just put all this in an interface and prefix all the names with USS, which might result in shorter, cleaner code. Will think a bit on the best way to organize this.

Reply:
The library is only meant for namespace yes. Do you think an interface is cleaner than a library? I'm unopinionated

Reply:
I think it's cleaner as long as we're cool with prefixing everything, yeah.

Reply:
Hm yeah I think its cleaner too. For now I've defined functions and events/structs in `USSSpokePoolInterface.sol`.

Currently the existing `SpokePool` file implements the new functions. I thought about making a new `USSSpokePool` that either extends the SpokePool or is extended by the SpokePool. However, this has tradeoffs:

## abstract contract `USSSpokePool` is `SpokePool`

Requires replacing a lot of `SpokePool` references in contracts like `Arbitrum_SpokePool`, `Ovm_SpokePool` etc. that all are expecting the extend the `SpokePool` but would now want to extend `USSSpokePool`. This isn't that bad but touches a lot of files.

## abstract contract `SpokePool` is `USSSpokePool`

This would require declaring a `USSSpokePool` abstract contract that extends a lot of the same interfaces that `SpokePool` extends like `NonReentrancyGuardUpgradeable` and `EIP712CrossChainUpgradeable` while also adding some declarations like `using SafeERC20Upgradeable for IERC20Upgradeable` and `using AddressLibUpgradeable for address`. This is a bit verbose but probably cleaner than the first option because it contains the extra code to one file, `USSSpokePool`. These changes would also apply to declaring `USSSpokePool` as a library.